src/utils/intersect.jl

```diff
@@ -0,0 +1,346 @@
+  for seg in Iterators.flatten((ℬₚ, ℳₚ))
+    BinaryTrees.insert!(ℛ, SweepSegment(seg, sweepline))
+  end
+end
+
+function _handlebottom!(bundle, ℛ, 𝒬, p, digits)
+  # bundle is sorted sequence, so the first segment is minimum
+  s′ = bundle[begin]
+  # element below s′
+  nsₗ, _ = !isnothing(s′) ? BinaryTrees.prevnext(ℛ, s′) : (nothing, nothing)
+  if !isnothing(nsₗ)
+    _newevent!(𝒬, _sweepline(s′), bundle, p, _segment(s′), _keyseg(nsₗ), digits)
+  end
+end
+
+function _handletop!(bundle, ℛ, 𝒬, p, digits)
+  # bundle is sorted sequence, so the last segment is maximum
+  s″ = bundle[end]
+  # element above s″
+  _, nsᵤ = !isnothing(s″) ? BinaryTrees.prevnext(ℛ, s″) : (nothing, nothing)
+  if !isnothing(nsᵤ)
+    _newevent!(𝒬, _sweepline(s″), bundle, p, _segment(s″), _keyseg(nsᵤ), digits)
+  end
+end
+
+# -----------------
+# HELPER FUNCTIONS
+# -----------------
+
+# function to add new events to the event queue as needed
+function _newevent!(𝒬, sweepline, bundle, p, seg₁, seg₂, digits)
+  intersection(Segment(seg₁), Segment(seg₂)) do I
+    t = type(I)
+    if t === Crossing || t === EdgeTouching
+      i = coordround(get(I), digits=digits)
+      if i ≈ p # helps with vertical+horizontal intersections
+        push!(bundle, SweepSegment(seg₂, sweepline))
+        push!(bundle, SweepSegment(seg₁, sweepline))
+      elseif i > p # add to 𝒬, update existing point if needed
+        _addpoint!(𝒬, i, [seg₁, seg₂], 3)
+      end
+    end
+  end
+end
+
+# compute the number of significant digits based on the segment type
+# this is used to determine the precision of the points
+function _digits(segments)
+  seg = first(segments)
+  ℒ = lentype(seg)
+  τ = ustrip(eps(ℒ))
+  round(Int, 0.8 * (-log10(τ))) # 0.8 is a heuristic to avoid numerical issues
+end
+
+# add an endpoint and corresponding segment to the event queue at a given position (1=start, 2=end, 3=crossing)
+function _addpoint!(𝒬, p, segs, pos)
+  U = Set{typeof(first(segs))} # set of segments
+  node = BinaryTrees.search(𝒬, p)
+  # updates or adds event point based on existing events in 𝒬
+  if !isnothing(node)
+    union!(BinaryTrees.value(node)[pos], segs)
+  else
+    # create a new event point with the value of (U(), U(), U()). pos determines whether inserted segments are starts or ends
+    vals = ntuple(i -> i == pos ? U(segs) : U(), 3) # (Starts, Ends, Crossings)
+    BinaryTrees.insert!(𝒬, p, vals)
+  end
+end
+
+# compute y bounds of the segments
+function _ybounds(segs)
+  # compute bounding box
+  bbox = boundingbox(segs)
+
+  # stretch bounding bbox
+  T = numtype(lentype(bbox))
+  sbox = bbox |> Stretch(T(1.05)) 
+
+  # extract y coordinate values
+  map(p -> coords(p).y, extrema(sbox))
+end
+
+# convenience function to get the segment from a our AVLNode{SweepSegment} structure
+_keyseg(ns) = _segment(BinaryTrees.key(ns))
+
+# handles  the degenerate case of trivial (0-length) segments
+_istrivial(seg) = seg[1] == seg[2]
```

Comment:
I am concerned that I am still reviewing dead code. This function isn't used anywhere.